controllers/util/k8sutil.go

```diff
@@ -166,3 +166,15 @@ func GenerateNameForNonResilientCephBlockPoolSC(initData *ocsv1.StorageCluster)
 	}
 	return fmt.Sprintf("%s-ceph-non-resilient-rbd", initData.Name)
 }
+
+func GetStorageClusterInNamespace(ctx context.Context, cl client.Client, namespace string) (*ocsv1.StorageCluster, error) {
+	storageClusterList := &ocsv1.StorageClusterList{}
+	err := cl.List(ctx, storageClusterList, client.InNamespace(namespace))
```

Comment:
Can you please cap this list operation to 1 result?

Reply:
I was going through https://github.com/red-hat-storage/ocs-operator/pull/2837, and from what I understood we can have multiple storageClusters, but the other storageCluster's phase will be ignored. Modified the function there to GetStorageClusterInNamespace

Reply:
You cannot have multi-storage clusters in a single namespace, and this lists only inside the namespace. So my ask adding `client.Count(1)` is meaningful

Reply:
We can have multple storageclusters inside a single namespace, Just that except one all else would be in phase ignored. Limiting the list to 1 result can easily give us the wrong result with a phasIgnored storageCluster. I would suggest listing all the items & then filtering from that list based on the requirement is the better approach.

Reply:
@malayparida2000 I understand what you are saying, but that is a scenario we should not address this way. That is a user error, and if we want to solve it we should introduce a webhook to prevent the creation of a second storage cluster. 

Listing with a count of (1) prevents the deep copy operations of all other storage classes from the cache into the resulting list, in the case of storage cluster it is a very very big structure (even if it is mostly not filled) and we should avoid the deep copy.

Reply:
@malayparida2000 @nb-ohad I will open an issue about it, so that we can create a webhook for it after the feature freeze, for now, we can list with a count of (1) and ignore the others

Reply:
Issue: https://github.com/red-hat-storage/ocs-operator/issues/2877

Reply:
pls excuse for pitching in and correct me if I'm not understanding it properly. Agreed, it's an user error but if it did happen, count of (1) has a min of 50% chance of going wrong.

until we make the storagecluster in a ns singleton I don't think we incur a lot of mem usage if runtime does a deep copy to provide us as the stack size can go upto MBs and last time I checked storagecluster is around 2KB (without managed fields) but w/ pointer and all it'll more than that but not in a 2 digit MBs.

however, I agree w/ Ohad since we are returning a pointer to caller and storagecluster escapes to heap so does the list that is queried in this function increasing GC and heap usage, but kube clients outside of manager aren't backed up by cache and so every list call hits the api server (with the limit if set).

imo, we should code for correctness and list all storageclusters to filter out ignored one's unless we make storagecluster a singleton before merging this PR.

Reply:
I completely agree with Leela. Whether or not we decide to enforce having a single StorageCluster as mandatory is a separate discussion. However, currently, limiting the results to just one could lead to incorrect outcomes if we actually have more than one StorageCluster. So we can not limit the result to count(1) as of now.

Regarding the idea of creating a webhook to prevent additional StorageCluster creation, let's consider the practical implications. In most cases, a customer would either have a single StorageCluster as intended, or they might mistakenly create a second one. It's unlikely that a customer would intentionally create hundreds of ignored StorageCluster CRs to overwhelm our cache. Given this, the overhead of implementing a webhook to restrict additional StorageCluster creation might outweigh listine an extra StorageCluster in the "ignored" phase if one exists due to a customer error.

Reply:
@leelavg The point is that the probability of that happening is very low and we will inform that on the log and will instruct the user/support to delete the other StorageCluster. The code as it is now just ignores the additional storage cluster which is the wrong thing to do! It should be an error. 

now, considering the amount of places and flows where this method is invoked and the count of deep copies this method will create for a very big struct (storage cluster). I am asking us to treat an additional storage cluster as an error not as thing that is just ignored

@rewantsoni please make that change as requested

Reply:
@malayparida2000 
> Given this, the overhead of implementing a webhook to restrict additional StorageCluster creation might outweigh listine an extra StorageCluster in the "ignored" phase if one exists due to a customer error.

We can deliberate about the nececity for a webhook at a later time, that has nothing to do with the fact that the system is misconfigured and we just ignoring it.